sympy/solvers/bivariate.py

```diff
@@ -114,6 +120,45 @@ def _linab(arg, symbol):
     return a, b, x
 
 
+def _power_list(f, *symbols):
+    """
+    Helper function to return powers of variables
+    present in f in list of dictionaries.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The expression in which powers of symbols present is to be returned.
+
+    symbols : Symbol
+        The symbols whose powers are to be returned.
+
+    Returns
+    =======
+
+    Powers of symbols in f in list of dictionaries form.
```

Comment:
[This](https://github.com/sympy/sympy/pull/16890#discussion_r297824748) comment is not addressed yet.
How would you know, what power is of what symbol?

Reply:
For that I had modified `_power_list` as
```
result = {}
    for t in term_factors(f):
        if isinstance(t, (trigs, Abs, log, exp)):
            # result += _power_list(t.args[0], *symbols)
        elif t.has(*symbols):
            p = t.as_poly()
            for i in symbols:
                if i in p.gens:
                    if i not in result.keys():
                        result[i] = flatten(p.monoms())
                    else:
                        result[i] += flatten(set(p.monoms()))
    return result
```
This is returning result in form of dictionary like {x:[powers of x present], y: [powers of y present]}
The only thing left is to merge these two dictionaries
` result += _power_list(t.args[0], *symbols)`
if you could help me in this.